pkg/storage/cloudimpl/s3_storage.go

```diff
@@ -175,11 +162,29 @@ func MakeS3Storage(
 		conf:     conf,
 		ioConf:   ioConf,
 		prefix:   conf.Prefix,
-		s3:       s3.New(sess),
+		opts:     opts,
 		settings: settings,
 	}, nil
 }
 
+func (s *s3Storage) newS3Client(ctx context.Context) (*s3.S3, error) {
+	sess, err := session.NewSessionWithOptions(s.opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "new aws session")
+	}
+	if s.conf.Region == "" {
+		if err := delayedRetry(ctx, func() error {
+			var err error
+			s.conf.Region, err = s3manager.GetBucketRegion(ctx, sess, s.conf.Bucket, "us-east-1")
```

Comment:
this only runs the first time, since it mutates s.conf.Region so it's memoized for future calls?

Reply:
Yup -- I really can't see region changing for a bucket.